### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by spencerahill**
This is part of our larger ongoing effort to massively overhaul the documentation, etc.  Rather than doing this as a single big PR, I think we should add components as we go, since they are so sorely needed.  This particular PR is for the top-level README, which seems to me a good place to start.

@spencerkclark could you take a look when you get a chance?  Anything I missed, or too long, typos, hard to understand, or anything else?  Thanks!

### CODE REVIEW

**Comment by spencerahill** [comment]
Actually, now that I see it actually rendered, I see that the first section has way too much text.  So maybe do a much shorter version there, and leave the more detailed explanation for the actual docs?

Also I see that my internal links don't work; will fix.

**Comment by spencerkclark** [comment]
@spencerahill I think this is a very well written introduction to aospy.  Thanks for putting this together!  

I agree we might want to move some portions to the docs to keep down the length on the repo splash page.  I'll put in my two cents for things we could consider moving in some line-comments, as well as address a couple typos.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+* Computing multiple statistical  (e.g. mean, standard deviation) and physical (e.g. column integrals or averages; zonal integrals or averages) reductions on any given computation, both on a gridpoint-by-gridpoint basis and over an arbitrary number of geographical regions
+* any combination of the above, plus many more!
+
+In fact, aospy enables the user to specify an arbitrary combination of models, simulations, variables to compute, time ranges, months of the year, input data types, output data types, and other parameters and to submit them all for execution with a single command.  The desired computations are then executed in parallel, and their outputs are stored as netCDF files in a highly structured directory format with metadata imbued within (1) the netCDF files themselves, (2) the filenames, and (3) the directory structure.  This prevents the all-to-common occurrence of computations becoming that get saved to disk becoming obsolete as time passes (or for other users), due to insufficient metadata describing precisely what they are and how they were computed.
```
Comment (spencerkclark) [inline README.md]: Typo: "This prevents the all-*too*-common occurrence..."

File: README.md [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+
+## Troubleshooting
+Questions of any kind are welcome and best placed as Issues on the Github repo.
+
+# Development
+## Contributions
+aospy is currently developed by Spencer Hill (@spencerahill) and Spencer Clark (@spencerkclark).  We are actively seeking new users and/or contributors, including during the ongoing refactor/documentation sprint.  Please get in touch however you'd like!
+
+## Copyright and license
+Copyright 2016, the aospy developers.  See [./LICENSE](the license) for further informaiton on acceptable usage, etc.
```
Comment (spencerkclark) [inline README.md]: Typo: "...for further *information*..."

File: README.md [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+* Computing multiple statistical  (e.g. mean, standard deviation) and physical (e.g. column integrals or averages; zonal integrals or averages) reductions on any given computation, both on a gridpoint-by-gridpoint basis and over an arbitrary number of geographical regions
+* any combination of the above, plus many more!
+
+In fact, aospy enables the user to specify an arbitrary combination of models, simulations, variables to compute, time ranges, months of the year, input data types, output data types, and other parameters and to submit them all for execution with a single command.  The desired computations are then executed in parallel, and their outputs are stored as netCDF files in a highly structured directory format with metadata imbued within (1) the netCDF files themselves, (2) the filenames, and (3) the directory structure.  This prevents the all-to-common occurrence of computations becoming that get saved to disk becoming obsolete as time passes (or for other users), due to insufficient metadata describing precisely what they are and how they were computed.
+
+## Design philosophy
```
Comment (spencerkclark) [inline README.md:18]: This is a subsection I think we could move to the docs

File: README.md [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+Key to enabling this automation and handling of model- and simulation-level idiosyncrasies is separating
+
+1. Code that describes the data that you want to work with
+2. Code that specifies any physical calculations you eventually want to perform
+3. Code that specifies the set of computations the user wishes to perform at a given time
+
+For (1), the user defines objects at three distinct levels: `Proj`, `Model`, and `Run`, that specify where the data is located that you want to work with.  For (2), the user defines `Var` objects that describe the physical quantities to be computed, including any functions that transform one or more directly model-outputted quantities into the ultimately desired quantity.  Once these objects have been defined, the user can proceed with (3) via a simple script that specifies any models, simulations, physical quantities, etc. to be performed.  aospy then permutes over the specified combinations of inputs and desired outputs, submits all the calculations in parallel, and then saves the output as netCDF files in a highly structured and metadata-infused directory and filename structure.
```
Comment (spencerkclark) [inline README.md]: The last sentence of this subsection:

> aospy then permutes over the specified combinations of inputs and desired outputs, submits all the calculations in parallel, and then saves the output as netCDF files in a highly structured and metadata-infused directory and filename structure.

is very similar to:

>In fact, aospy enables the user to specify an arbitrary combination of models, simulations, variables to compute, time ranges, months of the year, input data types, output data types, and other parameters and to submit them all for execution with a single command. The desired computations are then executed in parallel, and their outputs are stored as netCDF files in a highly structured directory format with metadata imbued within (1) the netCDF files themselves, (2) the filenames, and (3) the directory structure.

in the previous subsection.  I tend to like the prior subsection's version more.  Maybe consider removing the sentence here?

File: README.md (hunk 1 above)
Comment (spencerkclark) [inline README.md]: >The desired computations are then executed in parallel...

This is a bit of a nit, but technically ```aospy``` does not do permutation or parallelization; it is the way our current main.py script is set up in our object libraries that enables this.  Maybe rephrase this as "The desired computations *can then be* executed in parallel..."?

File: README.md [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+
+```
+git clone https://www.github.com/spencerahill/aospy/aospy.git
+cd aospy
+python setup.py install
+```
+
+Once we reach the v0.1 release, aospy will become available on PyPI (i.e. `pip install aospy`) and conda (`conda install aospy`).  For now, just use the above `git clone` method.  Please open an issue if you have any installation problems.
+
+## Performing computations
```
Comment (spencerkclark) [inline README.md]: For now maybe title this subsection simply "Examples"?

**Comment by spencerahill** [comment]
Thanks @spencerkclark!  I included those and also copied (most of) the new contents of the README to the documentation itself, since the latter was even less verbose.
